jvm-packages/xgboost4j/src/test/java/ml/dmlc/xgboost4j/java/DMatrixTest.java

```diff
@@ -57,6 +57,31 @@ public void testCreateFromDataIterator() throws XGBoostError {
     }
   }
 
+  @Test
+  public void testCreateFromDataIteratorWithDiffFeatureSize() throws XGBoostError {
+    //create DMatrix from DataIterator
+
+    java.util.ArrayList<Float> labelall = new java.util.ArrayList<Float>();
+    int nrep = 3000;
+    java.util.List<LabeledPoint> blist = new java.util.LinkedList<LabeledPoint>();
+    int featureSize = 4;
+    for (int i = 0; i < nrep; ++i) {
+      // set some rows with wrong feature size
+      if (i % 10 == 1) {
+        featureSize = 5;
+      }
+      LabeledPoint p = new LabeledPoint(
+        0.1f + i, featureSize, new int[]{0, 2, 3}, new float[]{3, 4, 5});
+      blist.add(p);
+      labelall.add(p.label());
+    }
+    try {
+      DMatrix dmat = new DMatrix(blist.iterator(), null);
+    } catch (XGBoostError e) {
+
+    }
```

Comment:
no matter DMatric will throw exception or not, it will always pass test, no?

Reply:
@CodingCat  Thx for your pointing. add assert